### ISSUE TITLE
Warning: data generated is from more than 5 days.

### ISSUE THREAD

**Issue by Thomieh73** [wdecoster/NanoPlot#211]
Hi, 

i am working with a dataset that consists of multiple runs. 
My workflow: 
Guppy basecalling and then qcat demultiplexing (creates 96 barcode fastq files) and then trimming with Nanofilt. Then I run Nanoplot on the output from Nanofilt, which creates an error. 

```

Warning: data generated is from more than 5 days.
Likely this indicates you are combining multiple runs.
Plots based on time are invalid and therefore truncated to first 5 days.
```

and it produces this error as well, but that is part of it.

```
Traceback (most recent call last):
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/bin/NanoPlot", line 8, in <module>
    sys.exit(main())
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplot/NanoPlot.py", line 101, in main
    plots = make_plots(datadf, settings)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplot/NanoPlot.py", line 204, in make_plots
    plot_settings=plot_settings)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplotter/timeplots.py", line 57, in time_plots
    plot_settings=plot_settings)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplotter/timeplots.py", line 63, in violin_plots_over_time
    dfs['timebin'] = add_time_bins(dfs)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplotter/timeplots.py", line 171, in add_time_bins
    bins=ceil((maxtime / 3600) / bin_length),
ValueError: cannot convert float NaN to integer
```

But this now crashes the pipeline, and I can not create any other plots. Maybe not too bad, but I do like the hmtl page output.

What is interesting for me, is that Nanoplot did not have this error when analyzing the summary file that came out of Guppy for this same dataset?

Any ideas on how to deal with this error?

**Comment by Thomieh73** [comment]
This was the last bit of my error log:

```
2020-10-13 17:01:18,140 Calculated statistics
2020-10-13 17:01:18,141 Using sequenced read lengths for plotting.
2020-10-13 17:01:18,422 Hidding 0 reads longer than 3000bp in length plots.
2020-10-13 17:01:18,423 NanoPlot:  Valid color #4CB391.
2020-10-13 17:01:18,423 NanoPlot:  Valid colormap Greens.
2020-10-13 17:01:18,424 NanoPlot:  Creating length plots for Read length.
2020-10-13 17:01:18,468 NanoPlot:  Using 62 reads maximum of 1363bp.
2020-10-13 17:03:09,678 Created length plots
2020-10-13 17:03:09,683 Time plots truncated to first 5 days: invalid timespan: nan days
2020-10-13 17:03:09,685 Nanoplotter: Creating timeplots using 0 reads.
2020-10-13 17:03:12,426 cannot convert float NaN to integer
Traceback (most recent call last):
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplot/NanoPlot.py", line 101, in main
    plots = make_plots(datadf, settings)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplot/NanoPlot.py", line 204, in make_plots
    plot_settings=plot_settings)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplotter/timeplots.py", line 57, in time_plots
    plot_settings=plot_settings)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplotter/timeplots.py", line 63, in violin_plots_over_time
    dfs['timebin'] = add_time_bins(dfs)
  File "/cluster/projects/nn9305k/src/miniconda/envs/nanoplot/lib/python3.7/site-packages/nanoplotter/timeplots.py", line 171, in add_time_bins
    bins=ceil((maxtime / 3600) / bin_length),
ValueError: cannot convert float NaN to integer
```

**Comment by wdecoster** [comment]
Interesting... I'll have a look. Thanks for reporting!

**Comment by Thomieh73** [comment]
Okay, I checked the output from Nanofilt, and I notice that most barcodes fastq files have no reads.  hmmmmm that is no good. 
rerunning nanofilt now. My q-score for the raw reads does not come above 10 with this dataset...

**Comment by wdecoster** [comment]
Q10 seems rather stringent, no?

**Comment by Thomieh73** [comment]
Well, I am trying to build experience here with this kind of data. So I was too optimistic with the quality :-D
median score is 8,1

**Comment by wdecoster** [comment]
We usually don't filter at all, but it probably depends on your application.

**Comment by Thomieh73** [comment]
It is a 16S rRNA amplicon dataset. I was thinking to at least do some quality filtering to throw out some of the really bad reads. That should reduce the diversity that we detect.

**Comment by wdecoster** [comment]
Oh well - I have no experience with that so I won't come up with recommendations. But okay that doesn't solve the error above.

**Comment by Thomieh73** [comment]
Okay, using less stringent settings for nanofilt gave more data to work with, but this error remained. I still got it.
What is weird is that this dataset was created by running a single minION flowcell for 72 hrs, which makes 3 days, not 5 days as indicated in the error.

**Comment by wdecoster** [comment]
The log message `Time plots truncated to first 5 days: invalid timespan: nan days` is also very odd. Would it be possible to share your input data? If not, could you try running with `--raw` and giving me that TSV output to attempt to reproduce?

**Comment by Thomieh73** [comment]
Hi sorry for not replying to you. I will look at this tomorrow and then try it out again.

**Comment by wdecoster** [comment]
Hi, do you have an update here?

**Comment by Thomieh73** [comment]
Hi,
I have an update for you. I figured out what caused this error. I could repeat it on several independent datasets.

I was using the following command:

``` 
NanoPlot -t 4 --fastq_minimal *.fastq  --plots hex dot --title Clean_data_Summary -o nanoplot_results/summary
```
Using the option --fastq_minimal causes this funky error.

There was no error when I used the command:

```
NanoPlot -t 4 --fastq *.fastq  --plots hex dot --title Clean_data_Summary -o nanoplot_results/summary
```

Sorry again for the slow response. I finally got the time to address this and look carefully at what I was doing.
